Automattic/wp-calypso
# Purchases: Fetch VAT vendor info from endpoints

Calypso uses the tax vendor info provided by the hard-coded function `getVatVendorInfo()` for two purposes: first, it uses the address and tax ID for displaying information on the billing history receipts that can be accessed on the `/me/purchases/billing` page; second, it uses the `tax_name` property to localize the word "VAT" in various places like the "Add VAT details" checkbox in checkout.

However, the information in `getVatVendorInfo()` is also duplicated on the backend since it needs that information to generate email receipts. Having two sources of truth means extra work to keep the two in sync.

## Proposed Changes

This PR replaces the hard-coded VAT vendor info with data returned by endpoints added in D111309-code. It does this by splitting up the two uses of the data.

For the `tax_name` property, the countries list endpoint now includes that property for each taxable country. Every page that needs the tax name can now get it from that endpoint.

For the billing history pages, the `/me/billing-history` endpoint now includes a `tax_vendor_info` property which has the `address` and `vat_id` properties needed to properly render a receipt.

Requires D111309-code

## Screenshots

A receipt page with vendor information (VAT):

<img width="731" alt="Screenshot 2023-05-18 at 6 50 56 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/4b976566-c93d-4427-81b0-ecab2c009d09">

Another receipt page with vendor information (GST):

<img width="729" alt="Screenshot 2023-05-18 at 6 51 44 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/82b2d2fc-9bdf-4954-8bd9-4a39995e8889">

Checkout with tax name (GST):

<img width="574" alt="Screenshot 2023-05-18 at 6 52 42 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/ce2a60d7-f853-449d-98f3-6676c386e12a">

## Testing Instructions

- First apply D111309-code and sandbox the API.
- To verify that checkout gets the correct tax name, start at `/plans` and add a product to your cart. Verify that when you change the billing country to various taxable countries, the name of the taxes change to the appropriate one (eg: Australia should be "GST", United Kingdom should be "VAT", Japan should be "CT").
- To verify that receipt pages have the correct vendor info, make a purchase setting the country code in checkout to a taxable country. Then visit `/me/purchases/billing` clicking through to the purchase. Verify that it includes a "Vendor..." section with an address and a tax ID (see the screenshots above for examples).
